tesser.hadoop: reduce in two stages

egamble opened the pull request.

aphyr commented on hadoop/src/tesser/hadoop.clj:
```diff
@@ -200,3 +200,68 @@
+  {::mr/source-as :vals
+   ::mr/sink-as   :vals}
+  [records]
+  (r/map identity records))
```
I think you can just return `records` here.

egamble replied:
I thought I tried that and it didn't work. Just tried it again and it worked, so I changed it.

aphyr commented on hadoop/src/tesser/hadoop.clj:
```diff
@@ -200,3 +200,68 @@
+  (r/map identity records))
+
+(defn partition-randomly
+  ^long [key val ^long nparts]
```
Can we get a docstring on this? Why are key and val ignored? Is ^long supposed to be a type hint on `partition-randomly`?

egamble replied:
key is null, the actual content of val is not necessary for assigning randomly. I changed key and val to underscores, is that better? Yes, most examples of partitioning functions I saw declared both the type of nparts and the return type of the partitioning function, to avoid boxing.

aphyr commented on hadoop/src/tesser/hadoop.clj:
```diff
@@ -200,3 +200,68 @@
+  ^long [key val ^long nparts]
+  (rand-int nparts))
+
+(defn set-one-reducer
```
Since this mutates the conf, should probably be named `set-one-reducer!`

aphyr commented:
This looks good to me, Evan. Couple minor questions and I think we'll be good to go. Are you getting good results in prod? If so, I'll cut straight to 1.0.1 and release for you. :)

egamble commented:
Yes, good results in production. Added doc string, changed identity-mapper as you suggested (thought I had found that didn't work, but it does in fact work), etc.

aphyr commented:
Oh, gotcha, the partitioner fun is conforming to another API! Yeah, looks good to me. Merging and releasing! :)
